### PR TITLE
feat/MSSDK-1421: [1421, 1422, 1423] - Add RedeemGift component

### PR DESCRIPTION
**Pull request by m4tewoosh**
### Description
https://cleeng.atlassian.net/browse/MSSDK-1421
https://cleeng.atlassian.net/browse/MSSDK-1422
https://cleeng.atlassian.net/browse/MSSDK-1423

### Updates
Added separate `RedeemGift` component

### Screenshots

image
image
image
image
image

### CODE REVIEW

**Review by kkarasek — APPROVED** [review]
Looks OK!

**Review by Tarnoruda — APPROVED** [review]

File: src/components/RedeemGift/RedeemGift.tsx [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+    return () => clearTimeout(timer);
+  }, [isGiftRedeemed]);
+
+  const isConfirmButtonDisabled =
+    !redeemable || isVerifyLoading || isOfferLoading;
+
+  return (
+    <WrapperStyled>
+      <Header onBackClick={onBackClick} />
+      {isGiftRedeemed ? (
```
Comment (jkorczakcleeng) [inline src/components/RedeemGift/RedeemGift.tsx:103]: I'd consider break this jsx into two separate component. One for isGiftRedeemed === true and another when it is false. It should improve readability.

File: src/redux/giftSlice.ts [hunk 2]
```diff
@@ -42,6 +45,39 @@ export const fetchUpdateGift = createAsyncThunk<
...
+export const fetchVerifyGift = createAsyncThunk<
+  VerifiedGift,
+  string,
+  {
+    rejectValue: string;
+  }
+>('gift/verifyGift', async (giftCode: string, { rejectWithValue }) => {
+  try {
+    const result = await verifyGift(giftCode);
+
```
Comment (jkorczakcleeng) [inline src/redux/giftSlice.ts:57]: remove empty line

File: src/components/OfferCheckoutCard/OfferCheckoutCard.tsx [hunk 3]
```diff
@@ -270,30 +326,38 @@ const OfferCheckoutCard = () => {
...
+          <SkeletonWrapper showChildren={!loading} width={80} height={30}>
+            {isTrialBadgeVisible && (
+              <TrialBadgeStyled>
+                {renderTrialBadgeDescription()}
+              </TrialBadgeStyled>
+            )}
+            <Price
+              currency={currencyFormat[currency]}
+              price={
+                discountType === 'coupon' && totalPrice !== 0
```
[the remark this answers is not in the document]
Comment (m4tewoosh) [reply src/components/OfferCheckoutCard/OfferCheckoutCard.tsx:348]: Yup, good idea

File: src/components/OfferCheckoutCard/OfferCheckoutCard.tsx [hunk 4]
```diff
@@ -270,30 +326,38 @@ const OfferCheckoutCard = () => {
...
+                  ? Number(totalPrice)
+                  : Number(grossPrice)
+              }
+              period={
+                offerType === 'S' && period !== 'season'
```
[the remark this answers is not in the document]
Comment (m4tewoosh) [reply src/components/OfferCheckoutCard/OfferCheckoutCard.tsx:353]: As above

File: src/components/RedeemGift/RedeemGift.tsx [hunk 5]
```diff
@@ -0,0 +1,178 @@
...
+    verifiedGift: { redeemable }
+  } = useAppSelector(selectGift);
+
+  const { loading: isOfferLoading } = useAppSelector(selectOffer);
+
+  const handleChange = (e: React.ChangeEvent<HTMLInputElement>) =>
+    setGiftCode(e.target.value);
+
+  const handleVerifyGift = async () => {
+    const resultVerifyGiftAction = await dispatch(fetchVerifyGift(giftCode))
```
[the remark this answers is not in the document]
Comment (m4tewoosh) [reply src/components/RedeemGift/RedeemGift.tsx:52]: Yup, also good idea, I tried it before but it crashed when verify failed (only error is returned), but I didn't have catch here yet. After adding catch forgot to destructure it 😀

File: src/components/RedeemGift/RedeemGift.tsx [hunk 6]
```diff
@@ -0,0 +1,178 @@
...
+            <OfferWrapperStyled>
+              <OfferCheckoutCard isRedeemGift />
+            </OfferWrapperStyled>
+          )}
+          <Button
+            disabled={isConfirmButtonDisabled}
+            theme="confirm"
+            onClickFn={handleRedeemGift}
+          >
+            {isRedeemLoading ? (
```
[the remark this answers is not in the document]
Comment (jkorczakcleeng) [reply src/components/RedeemGift/RedeemGift.tsx:135]: cool, lets keep it that way ;)

**Review by jkorczakcleeng — APPROVED** [review]

**Review by pawelacio — APPROVED** [review]

**Review by kkarasek — APPROVED** [review]

**Pull request merged**